lib/model-interfaces/bedrock-agents/index.ts

```diff
@@ -0,0 +1,120 @@
+import * as cdk from "aws-cdk-lib";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as lambdaEventSources from "aws-cdk-lib/aws-lambda-event-sources";
+import * as logs from "aws-cdk-lib/aws-logs";
+
+import * as sns from "aws-cdk-lib/aws-sns";
+import * as sqs from "aws-cdk-lib/aws-sqs";
+import { Construct } from "constructs";
+import * as path from "path";
+import { Shared } from "../../shared";
+import { Direction, ModelInterface, SystemConfig } from "../../shared/types";
+import { SqsSubscription } from "aws-cdk-lib/aws-sns-subscriptions";
+
+interface BedrockAgentInterfaceProps {
+  readonly shared: Shared;
+  readonly config: SystemConfig;
+  readonly messagesTopic: sns.Topic;
+  readonly sessionsTable: dynamodb.Table;
+  readonly byUserIdIndex: string;
+}
+
+export class BedrockAgentInterface extends Construct {
+  public readonly ingestionQueue: sqs.Queue;
+  public readonly requestHandler: lambda.Function;
+
+  constructor(scope: Construct, id: string, props: BedrockAgentInterfaceProps) {
+    super(scope, id);
+
+    const requestHandler = new lambda.Function(this, "RequestHandler", {
+      vpc: props.shared.vpc,
+      code: props.shared.sharedCode.bundleWithLambdaAsset(
+        path.join(__dirname, "./functions/request-handler")
+      ),
+      handler: "index.handler",
+      runtime: props.shared.pythonRuntime,
+      architecture: props.shared.lambdaArchitecture,
+      tracing: lambda.Tracing.ACTIVE,
+      timeout: cdk.Duration.minutes(15),
+      memorySize: 1024,
+      logRetention: logs.RetentionDays.ONE_WEEK,
```

Comment:
```suggestion
      logRetention: props.config.logRetention ?? logs.RetentionDays.ONE_WEEK,
       loggingFormat: lambda.LoggingFormat.JSON,
```

I recently added customizable retention in #552